### PR TITLE
Change status column to be 4 explicit ones

### PR DESCRIPTION
**Pull request by ofedoren**
@lzap, I've added migration only, since I'll change the way we show statuses when we have something coming from the proxy.

UPD: My bad, forgot to update few more things...

### CODE REVIEW

**Comment by ofedoren** [comment]
@lzap, updated. I've made all these columns as optional params since we presume those to be in the body for now. Not sure if we need to make it required. Please, review and share your concerns.

File: db/migrate/20211011141813_change_status_column.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class ChangeStatusColumn < ActiveRecord::Migration[6.0]
+  def change
+    remove_column :host_reports, :status, :bigint
+
+    add_column :host_reports, :passed, :integer
+    add_column :host_reports, :failed, :integer
+    add_column :host_reports, :pending, :integer
+    add_column :host_reports, :other, :integer
+  end
+end
```
Comment (lzap) [inline db/migrate/20211011141813_change_status_column.rb:10]: Could be `default: 0` I assume?

**Comment by lzap** [comment]
Could you please do PR into SP too? It is indeed missing there, but it should be easy to add it. There are fixtures with various examples.

**Comment by lzap** [comment]
We have discussed the naming, this is the proposal:

```
## Status mapping

### Puppet

* changed -> applied
* corrective_change -> applied
* failed -> failed
* failed_to_restart -> failed
* scheduled -> pending
* restarted -> other
* skipped -> other
* out_of_sync
* total

### Ansible

* applied -> applied
* failed -> failed
* skipped -> other
* pending -> pending
```

**Comment by ofedoren** [comment]
@lzap, updated per https://github.com/theforeman/smart_proxy_host_reports/pull/9.

File: test/controllers/api/v2/host_reports_controller_test.rb [hunk 2]
```diff
@@ -37,8 +37,7 @@ def test_create_valid
       User.current = nil
       post :create, params: {
         host_report: {
-          host: host.name, body: report_body, reported_at: Time.current,
-          status: 0
+          host: host.name, body: report_body, reported_at: Time.current
```
Comment (lzap) [inline test/controllers/api/v2/host_reports_controller_test.rb]: Perhaps one test that confirms it stores these columns correctly?

**Comment by lzap** [comment]
No other comments, I am going to test and merge once the SPPR is ready.

**Comment by ofedoren** [comment]
@lzap, updated.

**Comment by lzap** [comment]
It works fine, however, JS tests are failing. Please fix and I am merging.

**Comment by ofedoren** [comment]
@lzap, updated and green.

**Review by lzap — APPROVED** [review]
Works fine, merging thanks.

**Pull request merged**